src/BinSkim.Driver/RoslynAnalyzer/ILOperations.cs

```diff
@@ -0,0 +1,485 @@
+        public IExpression Operand { get; }
+        public override ITypeSymbol ResultType { get; }
+
+        public override OperationKind Kind => OperationKind.UnaryOperatorExpression;
+    }
+
+    internal sealed class BinaryOperatorExpression : HasOperatorExpression, IBinaryOperatorExpression
+    {
+        public BinaryOperatorExpression(BinaryOperationKind binaryKind, IExpression left, IExpression right, ITypeSymbol resultType)
+        {
+            BinaryKind = binaryKind;
+            Left = left;
+            Right = right;
+            ResultType = resultType;
+        }
+
+        public BinaryOperationKind BinaryKind { get; }
+        public IExpression Left { get; }
+        public IExpression Right { get; }
+        public override ITypeSymbol ResultType { get; }
+        public override OperationKind Kind => OperationKind.BinaryOperatorExpression;
+    }
+
+    internal sealed class ConversionExpression : HasOperatorExpression, IConversionExpression
+    {
+        public ConversionExpression(ConversionKind conversion, IExpression operand, ITypeSymbol resultType)
+        {
+            Conversion = conversion;
+            Operand = operand;
+        }
+
+        public ConversionKind Conversion { get; }
+        public IExpression Operand { get; }
+        public override ITypeSymbol ResultType { get; }
+        public override OperationKind Kind => OperationKind.ConversionExpression;
+
+        bool IConversionExpression.IsExplicit => true;
+    }
+
+
+    internal sealed class SizeOfExpression : Expression, ITypeOperationExpression
+    {
+        public SizeOfExpression(ISymbolProvider provider, ITypeSymbol typeOperand)
+        {
+            TypeOperand = typeOperand;
+            ResultType = provider.GetSpecialType(SpecialType.System_UInt32);
+        }
+
+        public ITypeSymbol TypeOperand { get; }
+        public override ITypeSymbol ResultType { get; }
+        public override OperationKind Kind => OperationKind.TypeOperationExpression;
+        public TypeOperationKind TypeOperationClass => TypeOperationKind.SizeOf; // FEEDBACK: Inconsistent naming of enum and member.
+    }
+
+    // TODO: TypeOfExpression should be here.
+    // FEEDBACK: It would be easier to express ldtoken directly, with separate GetTypeFomHandle call.
+    //           Also exposes gap for field and method handles.
+
+    internal sealed class LiteralExpression : Expression, ILiteralExpression
+    {
+        public LiteralExpression(object value, ITypeSymbol type)
+        {
+            ConstantValue = value;
+            ResultType = type;
+        }
+
+        public override object ConstantValue { get; }
+        public override ITypeSymbol ResultType { get; }
+        public override OperationKind Kind => OperationKind.LiteralExpression;
+
+        // TODO: Proper IL/round-trippable syntax.
+        // FEEDBACK: Does Spelling belong on ILiteralExpression? Why not get it from IOperation.Syntax?
```

Comment:
Fair question, and I don't have a definitive answer. I kind of hope to get rid of the Syntax property, and don't want to introduce dependencies on it where necessary.